### ISSUE TITLE
quality/self-comparison triggers on different dotted paths

### ISSUE THREAD

**Issue by EvilsPresley** [atelico/gdstyle#7]
### Minimal reproduction

```gdscript
extends Node

@export var whatever: Node

func _ready():
  if position == whatever.position:
    print("hello")
```

### Command you ran

gdstyle

### What gdstyle did

This one is similar to an issue you solved from my reddit comment:

> 739:50 warning comparing 'label' with itself [quality/self-comparison]
>       		if stored_goods != null and stored_goods.label == label:

(Don't tell me my class is 740 lines long, I know it's bad :) )

### What you expected

No warning

### gdstyle version

gdstyle 0.1.7

### Platform

macOS (Apple Silicon)

**Comment by w4nderlust** [comment]
Thanks for the clear repro, this was a real one. The self-comparison rule was only looking at the last identifier on each side, so `stored_goods.label == label` got read as `label == label`. It now compares the full dotted path on both sides, so that case stays quiet while genuine ones like `obj.foo == obj.foo` are still caught.

Fixed in #10, will go out in the next release. (And no comment on the line count.) Thanks for reporting it.
